src/main/java/hudson/plugins/ec2/EC2Cloud.java

```diff
@@ -1081,19 +1117,22 @@ public static URL checkEndPoint(String url) throws FormValidation {
     @CheckForNull
     private static SSHUserPrivateKey getSshCredential(String id, ItemGroup context){
 
-        SSHUserPrivateKey credential = CredentialsMatchers.firstOrNull(
-            CredentialsProvider.lookupCredentials(
-                SSHUserPrivateKey.class, // (1)
-                context,
-                null,
-                Collections.emptyList()),
-            CredentialsMatchers.withId(id));
+        if (!id.isEmpty()) {
```

Comment:
I guess `id` is implicitly `@NonNull` here? Callers that I could see in this PR at least are checking for null first. Some are also checking `isEmpty`. A bit confusing; maybe review call sites and document the method appropriately.